util/conditions/setter.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"fmt"
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+)
+
+// Setter interface defines methods that a Cluster API object should implement in order to
+// use the conditions package for setting conditions.
+type Setter interface {
+	Getter
+	SetConditions(clusterv1.Conditions)
+}
+
+// Set sets the given condition.
+// Please note that a condition is set only when a state transition happens, where a state transition
+// is define by a change of one of the following field: Status, Reason, Severity and Message.
+// During a state transition, the LastTransitionTime is automatically set.
```

Comment:
```suggestion
// Set sets the given condition.
//
// Note: A condition should be set only when a state transition happens. A state transition
// is a change of one of the following field: Status, Reason, Severity and Message.
// During a state transition, the LastTransitionTime is automatically set.
```

The last 3 lines are a little confusing, what are we trying to say?

Reply:
I have updated the comment, PTAL